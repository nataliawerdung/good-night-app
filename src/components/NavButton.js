import React, { Component } from 'react'
import styled from 'react-emotion'
import { Link } from 'react-router-dom'
import { css } from 'emotion'

const StyledDiv = styled('div')`
  grid-area: nav;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
`
const Button = styled('div')`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 15px;
  background: whitesmoke;
  border-radius: 7px;
  height: 55px;
  width: 105px;
  border: none;
  transition: all 0.4s ease;

  &:hover {
    transform: scale(1.1);
  }
`
const Span = styled('div')`
  font-size: 22px;
  color: rgb(29, 54, 73);
  margin: 5px;
`

const noUnderline = css`
  text-decoration: none;
`

export default class NavButton extends Component {
  render() {
    return (
      <StyledDiv>
        <Button>
          <Link className={noUnderline} to="/settings">
            <Span>settings</Span>
          </Link>
        </Button>
        <Button>
          <Link className={noUnderline} to="/statistics">
            <Span>statistics</Span>
          </Link>
        </Button>
      </StyledDiv>
    )
  }
}
