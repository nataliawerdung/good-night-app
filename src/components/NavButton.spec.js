import NavButton from './NavButton'
import { shallow } from 'enzyme'
import React from 'react'
import toJson from 'enzyme-to-json'

describe('NavButton', () => {
  let wrapper

  let mockFunction = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<NavButton onClick={mockFunction} />)
  })

  it('renders a NavButton', () => {
    expect(wrapper.html()).toContain('save')
  })

  it('has a matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('reacts on click', () => {
    expect(mockFunction.mock.calls.length).toBe(0)
    wrapper.find('button').simulate('click')
    expect(mockFunction.mock.calls.length).toBe(1)
  })
})
