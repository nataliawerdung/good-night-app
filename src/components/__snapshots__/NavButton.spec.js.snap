// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButton has a matching snapshot 1`] = `undefined`;
